torchtune/config/_utils.py

```diff
@@ -148,3 +150,37 @@ def _merge_yaml_and_cli_args(yaml_args: Namespace, cli_args: List[str]) -> DictC
 
     # CLI takes precedence over yaml args
     return OmegaConf.merge(yaml_conf, cli_conf)
+
+
+def _get_template(template: str) -> PromptTemplate:
+    """
+    Get the prompt template class from the template string.
+
+    String should either be the PromptTemplate class name directly, or a raw
+    string with 1 or more placeholders. If none of these apply, then raise an
+    error.
+
+    Args:
+        template (str): class name of template, or string with placeholders
+
+    Returns:
+        PromptTemplate: the prompt template class or the same verified string
+
+    Raises:
+        ValueError: if the template is not a PromptTemplate class or a proper
+            template string
+    """
+    path = "torchtune.data." + template
+    try:
+        template_class = _get_component_from_path(path)
+        return template_class()
+    except InstantiationError:
+        # Verify that string can be used as a template, should have variable
+        # placeholders
+        pattern = r"\{.+?\}"
```

Comment:
Is this the most robust validation? E.g. I think `\{hello\}` will pass but is not a valid template. Not to mention that we are not validating # of args or anything like that. Not a huge deal cause I know config validation is hard, but just wanna be realistic about how much we can accomplish with this.

Reply:
why is `{hello}` not a valid template? It technically is with the variable placeholder `hello`

Reply:
ok I think I just needed to tighten the regex:
<img width="416" alt="image" src="https://github.com/pytorch/torchtune/assets/33648637/05a60752-252b-4da6-9c84-6e7bc7d0a1fa">